src/layertreemodel.cpp

```diff
@@ -103,8 +109,19 @@ bool LayerTreeModel::setData(const QModelIndex& index, const QVariant& value, in
 {
   if ( role == Visible )
   {
-    QgsLayerTreeNode* node = mLayerTreeModel->index2node( mapToSource( index ) );
-    node->setItemVisibilityCheckedRecursive( value.toBool() );
+    if ( QgsLayerTreeModelLegendNode *sym = mLayerTreeModel->index2legendNode( mapToSource( index ) ) )
+    {
+      QVariant checked = Qt::Unchecked;
+      if( value.toBool() )
+        checked=Qt::Checked;
```

Comment:
the code would a little nicer IMHO with

```
Qt::CheckState checked = (Qt::CheckState) value.toBool();
sym->setData( checked, Qt::CheckStateRole );
```

Reply:
But the `Qt::CheckState` has to be `Qt::Checked` what is like 2..

Reply:
and it convert it to PartiallyChecked if true?

Reply:
yes

Reply:
too bad
```
Qt::CheckState checked = value.toBool() ? Qt::CheckState : Qt::UnChecked ;
```
but it makes no difference, sorry for the noise

Reply:
Thanks. Agree on your inputs anyway. Will make the code cuter.